cmake/modules/extensions.cmake

```diff
@@ -1269,9 +1269,19 @@ function(zephyr_linker_sources location)
   elseif("${location}" STREQUAL "NOCACHE_SECTION")
     set(snippet_path "${nocache_path}")
   elseif("${location}" STREQUAL "ITCM_SECTION")
-    set(snippet_path "${itcm_path}")
+    dt_has_chosen(HAS_ITCM PROPERTY "zephyr,itcm")
+    if(HAS_ITCM)
+      set(snippet_path "${itcm_path}")
+    else()
+      set(snippet_path "${rom_start_path}")
+    endif()
```

Comment:
caller of `zephyr_linker_sources(ITCM_SECTION ...)` has requested snippet to be placed inside `ITCM` but may suddenly get things placed in regular rom.

This I don't like, it should be more predictable for the caller where things are placed in the linker script.

Furthermore, this very surprising behavior is not documented.

Reply:
I wasn't entirely happy with this either. I have changed it now to a `FATAL_ERROR`.